server/service/appconfig.go

```diff
@@ -572,7 +572,9 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte, applyOpts fle
 			return nil, fleet.NewInvalidArgumentError("UI GitOps Mode: ", "Repository URL is required when GitOps mode is enabled")
 		}
 	}
-	appConfig.UIGitOpsMode = newAppConfig.UIGitOpsMode
+	if appConfig.UIGitOpsMode.GitopsModeEnabled || newAppConfig.UIGitOpsMode.GitopsModeEnabled {
```

Comment:
This condition is tricky. As it stands, this first condition, `appConfig.UIGitOpsMode.GitOpsModeEnabled` causes the default empty values to be set when GOM is currently enabled even if it's not included in the payload, meaning it gets disabled. So if you enable GitOps mode, then change another setting (e.g `mdm.mcaos_migration`, which you can still access in the UI when GitOps mode is on), GitOps mode gets turned off and the repository URL cleared.

Reply:
I think the condition we want is

`newAppConfig.UIGitOpsMode.GitOpsModeEnabled || (newAppConfig.UIGitOpsModeEnabled && !appConfig.UIGitOpsModeEnabled`

because:

<img width="1800" alt="Screenshot 2025-03-26 at 5 08 26 PM" src="https://github.com/user-attachments/assets/53abfe9b-0fe1-4084-86c4-4716b105ed76" />